mobile/src/actions/mediaBackup.js

```diff
@@ -39,36 +42,96 @@ async function getDirID (dir) {
+      backupAllowed(getState().mobile.settings.wifiOnly)
+    )
+  }
+
   dispatch(startMediaUpload())
-  if (force || (getState().mobile.settings.backupImages && backupAllowed(getState().mobile.settings.wifiOnly))) {
+
+  if (!await isAuthorized()) {
+    // force is only possible if the authorization is accepted
+    force = await updateValueAfterRequestAuthorization(force)
+    // disable backupImages when authorization is refused
+    if (getState().mobile.settings.backupImages && !force) {
+      await dispatch(setBackupImages(false))
+    }
+  }
+
+  if (canStart(force, getState)) {
     const photosOnDevice = await getFilteredPhotos()
     const alreadyUploaded = getState().mobile.mediaBackup.uploaded
     const photosToUpload = photosOnDevice.filter(photo => !alreadyUploaded.includes(photo.id))
     const dirID = await getDirID(dir)
     const totalUpload = photosToUpload.length
     let uploadCounter = 0
     for (const photo of photosToUpload) {
-      if (getState().mobile.mediaBackup.cancelMediaBackup) {
+      if (getState().mobile.mediaBackup.cancelMediaBackup || !canStart(force, getState)) {
         break
       }
-      if (backupAllowed(getState().mobile.settings.wifiOnly)) {
-        dispatch(currentUploading(photo, uploadCounter++, totalUpload))
-        const blob = await getBlob(photo)
-        const options = {
-          dirID,
-          name: photo.fileName
-        }
-        await cozy.client.files.create(blob, options).then(() => {
-          dispatch(successMediaUpload(photo))
-        }).catch(err => {
-          if (err.status === HTTP_CODE_CONFLICT) {
-            dispatch(successMediaUpload(photo))
-          }
-          console.log(err)
-        })
-      }
+      dispatch(currentUploading(photo, uploadCounter++, totalUpload))
+      await dispatch(uploadPhoto(dirID, photo))
     }
   }
+
   dispatch(endMediaUpload())
 }
+
+const uploadPhoto = (dirID, photo) => async (dispatch, getState) => {
+  const blob = await getBlob(photo)
+  const options = {
+    dirID,
+    name: photo.fileName
+  }
+  await cozy.client.files.create(blob, options).then(() => {
+    dispatch(successMediaUpload(photo))
+  }).catch(err => {
+    if (err.status === HTTP_CODE_CONFLICT) {
+      dispatch(successMediaUpload(photo))
+    }
+    console.log(err)
+  })
+}
+
+// backupImages
+
+export const backupImages = backupImages => async (dispatch, getState) => {
+  if (backupImages === undefined) {
+    backupImages = getState().mobile.settings.backupImages
+  }
+  backupImages = await updateValueAfterRequestAuthorization(backupImages)
+  await dispatch(setBackupImages(backupImages))
+
+  backupImagesAnalytics(backupImages, getState)
+  updateStatusBackgroundService(backupImages)
+  if (backupImages) {
+    dispatch(startMediaBackup(getMediaFolderName()))
+  }
+
+  return backupImages
+}
+
+const updateValueAfterRequestAuthorization = async (value) => {
+  if (value) {
+    value = await requestAuthorization()
```

Comment:
You should pass `requestAuthorization` function as a parameter if this function depends on a specific plugin.